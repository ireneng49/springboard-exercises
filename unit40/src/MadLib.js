import React, { useState } from "react";
import stories from "./stories";


const MadLib = ({ madlibValues, setDisplayLib }) => {

    const handleReset = () => {
        setDisplayLib(false);
    }

    const { noun, noun2, color, adjective } = madlibValues
    let { story } = madlibValues
    if (story === '') story = 'Once Upon a Time'

    const selectStory = () => {
        for (let i = 0; i < stories.length; i++) {
            if (stories[i].title === story) {
                return createStory(stories[i].text)
            }
        }
    }

    const createStory = (foundStory) => {
        let modifiedStory;
        modifiedStory = foundStory.replace('noun1', noun)
        modifiedStory = modifiedStory.replace('noun2', noun2)
        modifiedStory = modifiedStory.replace('color', color)
        modifiedStory = modifiedStory.replace('adj', adjective)
        return modifiedStory
    }

    const buttonStyle = {
        backgroundColor: 'lightgreen',
        borderRadius: '5px',
        padding: '5px 10px',
        fontWeight: '700',
        border: '2px solid black'
    }

    return (
        <div className="MadLib">
            <h3>{`Story: ${story}`}</h3>
            <h3>{selectStory()}</h3>
            <button style={buttonStyle} onClick={handleReset}>Make Another Story</button>
        </div>
    );
}

export default MadLib;