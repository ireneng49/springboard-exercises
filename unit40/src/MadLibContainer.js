import React, { useState } from "react";
import MadLibForm from "./MadLibForm";
import MadLib from "./MadLib";
import './Container.css'

const MadLibContainer = () => {
    const [displayLib, setDisplayLib] = useState(false);
    const [madlibValues, setMadlibValues] = useState(null);

    return (
        <div className="Container">
            <h1>Mad Libs!</h1>
            {displayLib ? <MadLib
                madlibValues={madlibValues}
                setMadlibValues={setMadlibValues}
                setDisplayLib={setDisplayLib}/>
                : <MadLibForm
                    setDisplayLib={setDisplayLib}
                    setMadlibValues={setMadlibValues}/>}


        </div>
    );
}

export default MadLibContainer;