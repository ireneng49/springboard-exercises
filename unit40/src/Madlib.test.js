import React, { useState } from "react";
import { fireEvent, render } from "@testing-library/react";
import MadLib from "./MadLib";
// importing parent comp doesn't seem to help with func defs
import MadLibContainer from "./MadLibContainer";

const madlibValues = {
    adjective: 'scaly',
    color: 'pink',
    noun: 'fish',
    noun2: 'dog',
    story: 'True Love'
}

// const [displayLib, setDisplayLib] = useState(false);
// const [madlibValues, setMadlibValues] = useState(null);

it("renders without crashing", function () {
    render(<MadLib madlibValues={madlibValues} />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<MadLib madlibValues={madlibValues} />);
    expect(asFragment()).toMatchSnapshot();
});

it("Should create correct MadLib", function () {
    const madlib = render(<MadLib madlibValues={madlibValues} />);

    expect(madlib.getByText('There was a pink fish who loved a scaly dog.')).toBeInTheDocument();
    expect(madlib.getByText('Story: True Love')).toBeInTheDocument();
    expect(madlib.getByText('Make Another Story')).toBeInTheDocument();
})

// should I be doing this testing from inside MadlibContainer?
// it("Should return to creation form upon clicking reset button", function () {
//     const madlib = render(<MadLib madlibValues={madlibValues}
//         setMadlibValues={setMadlibValues}
//         setDisplayLib={setDisplayLib} />);

//     expect(madlib.getByText('There was a pink fish who loved a scaly dog.')).toBeInTheDocument();

//     const resetButton = madlib.getByText('Make Another Story')
//     expect(resetButton).toBeInTheDocument()
//     fireEvent.click(resetButton)
//     // expect(madlib.queryByText('There was a pink fish who loved a scaly dog.')).not.toBeVisible();
// })