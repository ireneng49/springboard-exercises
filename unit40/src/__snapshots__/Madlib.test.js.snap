// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MadLib"
  >
    <h3>
      Story: True Love
    </h3>
    <h3>
      There was a pink fish who loved a scaly dog.
    </h3>
    <button
      style="background-color: lightgreen; border-radius: 5px; padding: 5px 10px; font-weight: 700; border: 2px solid black;"
    >
      Make Another Story
    </button>
  </div>
</DocumentFragment>
`;
